lib/report_plugins/daytoday.js

```diff
@@ -597,6 +620,129 @@ daytoday.report = function report_daytoday(datastorage,sorteddaystoshow,options)
+      $('#daytodaystatchart-' + day).append(table);
+
+      var chartData = [
+        { label: translate('Basal'), count: totalBasalInsulin, pct: (totalBasalInsulin / totalDailyInsulin * 100).toFixed(0) }, 
+        { label: translate('Bolus'), count: bolusInsulin, pct: (bolusInsulin / totalDailyInsulin * 100).toFixed(0) }
+      ];
+
+      // Insulin distribution chart
+      var width = 120;
+      var height = 120;
+      var radius = Math.min(width, height) / 2;
+
+      var color = d3.scale.ordinal().range([basalcolor, boluscolor]); 
+      
+      var labelArc = d3.svg.arc()
+        .outerRadius(radius / 2)
+        .innerRadius(radius / 2);
+
+      var svg = d3.select('#daytodaystatinsulinpiechart-' + day)
+        .append('svg')
+        .attr('width', width)
+        .attr('height', height)
+        .append('g')
+        .attr('transform', 'translate(' + (width / 2) + 
+          ',' + (height / 2) + ')');
+
+      var arc = d3.svg.arc()
+        .outerRadius(radius);
+
+      var pie = d3.layout.pie()
+        .value(function(d) { return d.count; })
+        .sort(null);
+
+      var g = svg.selectAll('.insulinarc')
+        .data(pie(chartData))
+        .enter()
+        .append('g')
+        .attr('class', 'insulinarc');
+      
+      g.append('path')
+        .attr('d', arc)
+        .attr('opacity', '0.5')
+        .attr('fill', function(d, i) { 
+          return color(d.data.label);
+        });
+        
+      g.append('text')
+        .attr('transform', function(d) { return 'translate(' + labelArc.centroid(d) + ')'; })
+        .attr("dy", ".15em")
+        .style('font-weight', 'bold')
+        .attr('text-anchor', 'middle')
+        .text(function(d) { 
+          return d.data.pct + '%'; 
+        });
+        
+      // Carbs pie chart
+      
+      var carbscolor = d3.scale.ordinal().range(['red']); 
+
+      var carbsData = [
+        { label: translate('Carbs'), count: data.dailyCarbs } 
+      ];
+      
+      var carbssvg = d3.select('#daytodaystatcarbspiechart-' + day)
+        .append('svg')
+        .attr('width', width)
+        .attr('height', height)
+        .append('g')
+        .attr('transform', 'translate(' + (width / 2) + 
+          ',' + (height / 2) + ')');
+
+      var carbsarc = d3.svg.arc()
+        .outerRadius(radius * data.dailyCarbs / options.maxDailyCarbsValue);
+
+      var pie = d3.layout.pie()
+        .value(function(d) { return d.count; })
+        .sort(null);
+
+      var carbsg = carbssvg.selectAll('.carbsarc')
+        .data(pie(carbsData))
+        .enter()
+        .append('g')
+        .attr('class', 'carbsarc');
+      
+      carbsg.append('path')
+        .attr('d', carbsarc)
+        .attr('opacity', '0.5')
+        .attr('fill', function(d, i) { 
+          return carbscolor(d.data.label);
+        });
+        
+      carbsg.append('text')
+        .attr('transform', function(d) { return 'translate(0,0)'; })
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["d" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2648873504/issues/source?bid=2452379&fileBranchId=3323621#l737)